File: frontend/src/services/reports.ts
/**
 * 报告中心 service
 *
 * 后端为 backend/app/api/routes/reports.py（前缀 /reports，
 * 经 next.config.js rewrite 由 /api/reports 代理）。
 *
 * 两个此前的契约错配已修正：
 * 1. 列表接口的参数是 type/page/page_size，不是 limit —— 传 limit 会被 FastAPI 忽略
 * 2. "最新报告"有专门端点 /reports/latest（返回 {report}），
 *    此前用 /reports?limit=1 打列表接口再读 .report，恒为 undefined，
 *    导致报告中心顶部的 KPI 卡片从未显示过
 */

/** KPI 摘要（由 workflow 生成，字段可能缺失） */
export interface ReportKpiSummary {
  total_products?: number
  alert_count?: number
  sales_records?: number
  report_date?: string
}

/** 报告列表项 */
export interface DailyReportSummary {
  id: string
  report_date: string
  report_type: string
  status: string
  kpi_summary: ReportKpiSummary | null
  trace_id: string | null
  created_at: string
}

/** 报告详情：列表项 + 完整 markdown 正文 */
export interface DailyReportDetail extends DailyReportSummary {
  report_content: string
}

const BASE = '/api/reports'

/** 失败时抛错，让调用方能区分"没有报告"和"接口挂了" */
async function api<T>(url: string): Promise<T> {
  const res = await fetch(url)
  if (!res.ok) {
    const detail = await res.text().catch(() => '')
    throw new Error(`GET ${url} 失败 (${res.status})${detail ? `: ${detail.slice(0, 200)}` : ''}`)
  }
  return res.json() as Promise<T>
}

export const reportService = {
  /** 报告列表（按类型 + 分页） */
  getReports: (params: { type: string; page?: number; pageSize?: number }) => {
    const q = new URLSearchParams({
      type: params.type,
      page: String(params.page ?? 1),
      page_size: String(params.pageSize ?? 20),
    })
    return api<{ reports: DailyReportSummary[]; page: number; page_size: number }>(`${BASE}?${q}`)
  },

  /** 最新一条报告（含完整正文 + KPI 摘要） */
  getLatestReport: (type: string) =>
    api<{ report: DailyReportDetail | null }>(`${BASE}/latest?type=${encodeURIComponent(type)}`),

  /** 报告详情 */
  getReport: (id: string) => api<{ report: DailyReportDetail }>(`${BASE}/${encodeURIComponent(id)}`),
}

File: backend/app/api/routes/reports.py
"""app/api/routes/reports.py — 报告中心 API

端点：
- GET /api/reports             报告列表（按 type 过滤）
- GET /api/reports/latest      最新报告（含完整 content）
- GET /api/reports/{report_id} 报告详情
"""
from __future__ import annotations

from typing import Any

from fastapi import APIRouter, HTTPException, Query

from backend.seed.demo.runner import get_daily_report_store

router = APIRouter(prefix="/reports", tags=["报告中心"])


@router.get("")
async def list_reports(
    type: str = Query(default="daily_report"),
    page: int = Query(default=1, ge=1),
    page_size: int = Query(default=20, ge=1, le=100),
) -> dict[str, Any]:
    """报告列表（按类型 + 分页）"""
    store = get_daily_report_store()
    reports = store.list(report_type=type, page=page, page_size=page_size)
    return {"reports": reports, "page": page, "page_size": page_size}


@router.get("/latest")
async def get_latest_report(
    type: str = Query(default="daily_report"),
) -> dict[str, Any]:
    """最新一条报告（含完整内容 + KPI 摘要）"""
    store = get_daily_report_store()
    report = store.get_latest(report_type=type)
    return {"report": report}


@router.get("/{report_id}")
async def get_report(report_id: str) -> dict[str, Any]:
    """报告详情"""
    store = get_daily_report_store()
    report = store.get(report_id)
    if report is None:
        raise HTTPException(status_code=404, detail=f"report {report_id} not found")
    return {"report": report}
